test/spl/degree_elevation_test.cc

```diff
@@ -0,0 +1,516 @@
+  ASSERT_THAT(after_elevation_->GetDegree(0).get(), original_->GetDegree(0).get() + 1);
+}
+
+TEST_F(BSplineFig5_35, HasMoreKnots) {  // NOLINT
+  ASSERT_THAT(after_elevation_->GetKnotVector(0)->GetNumberOfDifferentKnots(),
+              original_->GetKnotVector(0)->GetNumberOfDifferentKnots());
+  ASSERT_THAT(after_elevation_->GetKnotVector(0)->GetNumberOfKnots(), original_->GetKnotVector(0)->GetNumberOfKnots()
+      + original_->GetKnotVector(0)->GetNumberOfDifferentKnots());
+}
+
+TEST_F(BSplineFig5_35, HasMoreControlPoints) {  // NOLINT
+  ASSERT_THAT(after_elevation_->GetNumberOfControlPoints(), original_->GetNumberOfControlPoints()
+      + original_->GetKnotVector(0)->GetNumberOfDifferentKnots() - 1);
+}
+
+TEST_F(BSplineFig5_35, DoesNotChangeGeometricallyAfterDegreeElevation) {  // NOLINT
+  ASSERT_THAT(after_elevation_->AreGeometricallyEqual(*original_), true);
+}
+
+class ALinearNURBS : public Test {  // NOLINT
+ public:
+  ALinearNURBS() {
+    std::array<Degree, 1> degree = {Degree{1}};
+    KnotVectors<1> knot_vector_before = {std::make_shared<baf::KnotVector>(
+        baf::KnotVector({ParamCoord{0}, ParamCoord{0}, ParamCoord{0.3}, ParamCoord{0.6}, ParamCoord{1},
+                         ParamCoord{1}}))};
+    std::vector<baf::ControlPoint> control_points = {
+        baf::ControlPoint(std::vector<double>({1.0, 1.0})),
+        baf::ControlPoint(std::vector<double>({1.0, 2.0})),
+        baf::ControlPoint(std::vector<double>({2.0, 2.0})),
+        baf::ControlPoint(std::vector<double>({2.0, 3.0}))
+    };
+    std::vector<double> weights = {1.0, 1.5, 0.5, 1.0};
+    original_ = std::make_shared<spl::NURBS<1>>(knot_vector_before, degree, control_points, weights);
+    after_elevation_ = std::make_shared<spl::NURBS<1>>(*original_);
+    after_elevation_->ElevateDegree(0);
+  }
+
+ protected:
+  std::shared_ptr<spl::NURBS<1>> original_;
+  std::shared_ptr<spl::NURBS<1>> after_elevation_;
+};
+
+TEST_F(ALinearNURBS, HasELevatedDegree) {  // NOLINT
+  ASSERT_THAT(after_elevation_->GetDegree(0).get(), original_->GetDegree(0).get() + 1);
+}
+
+TEST_F(ALinearNURBS, HasMoreKnots) {  // NOLINT
+  ASSERT_THAT(after_elevation_->GetKnotVector(0)->GetNumberOfDifferentKnots(),
+              original_->GetKnotVector(0)->GetNumberOfDifferentKnots());
+  ASSERT_THAT(after_elevation_->GetKnotVector(0)->GetNumberOfKnots(), original_->GetKnotVector(0)->GetNumberOfKnots()
+      + original_->GetKnotVector(0)->GetNumberOfDifferentKnots());
+}
+
+TEST_F(ALinearNURBS, HasMoreControlPoints) {  // NOLINT
+  ASSERT_THAT(after_elevation_->GetNumberOfControlPoints(), original_->GetNumberOfControlPoints()
+      + original_->GetKnotVector(0)->GetNumberOfDifferentKnots() - 1);
+}
+
+TEST_F(ALinearNURBS, DoesNotChangeGeometricallyAfterDegreeElevation) {  // NOLINT
+  ASSERT_THAT(after_elevation_->AreGeometricallyEqual(*original_), true);
+}
+
+class Random1DBSplineForDegreeElevation : public Test {  // NOLINT
```

Comment:
Tests exactly the same functionality as before? Decide whether either deterministic or random test case could be removed.

Reply:
Is it okay to use deterministic tests for 1d and 2d and random tests for 3d?

Reply:
Maybe we should only have one random test for 1D and apart from this only deterministic tests for all dimensions?